shared/util.py

```diff
@@ -1648,4 +1650,87 @@ def trigger_indexer_with_retry(indexer_name: str, blob_name: str) -> bool:
         logging.error(f"[blob_trigger] Error in async retry logic: {str(e)}")
         return False
     finally:
-        loop.close()
\ No newline at end of file
+        loop.close()
+
+
+async def get_report_job(job_id: str, organization_id: str) -> Optional[Dict[str, Any]]:
+    """
+    Retrieve a report job from Cosmos DB.
+    
+    Args:
+        job_id: Unique identifier for the job
+        organization_id: Organization ID (used as partition key)
+        
+    Returns:
+        Job document or None if not found
+    """
+    try:
+        bsc = await get_blob_service_client()
+        container_client = bsc.get_container_client(container_name="reportJobs")
+        
+        job = container_client.read_item(item=job_id, partition_key=organization_id)
+        return job
+        
+    except CosmosResourceNotFoundError:
+        logging.warning(f"Report job {job_id} not found for organization {organization_id}")
+        return None
+    except Exception as e:
+        logging.error(f"Error retrieving report job {job_id}: {str(e)}")
+        return None
+
+async def update_report_job_status(
+    job_id: str, 
+    organization_id: str, 
+    status: str, 
+    result_metadata: Optional[Dict[str, Any]] = None,
+    error_payload: Optional[Dict[str, Any]] = None
+) -> bool:
+    """
+    Update the status and result of a report job.
+    
+    Args:
+        job_id: Unique identifier for the job
+        organization_id: Organization ID (used as partition key)
+        status: New status (QUEUED, RUNNING, SUCCEEDED, FAILED)
+        result_metadata: Metadata about the generated report (for SUCCEEDED status)
+        error_payload: Error information (for FAILED status)
+        
+    Returns:
+        True if update successful, False otherwise
+    """
+    try:
+        bsc = await get_blob_service_client()
+        container_client = bsc.get_container_client(container_name="reportJobs")
+        
+        # Get the current job
+        job = container_client.read_item(item=job_id, partition_key=organization_id)
```

Comment:
@Alexjes98 how does it work? how the blob client is able to read data from the cosmo container  reportJobs and the container name is acctually `report_jobs`:
<img width="840" height="80" alt="image" src="https://github.com/user-attachments/assets/e40b17b8-2dea-4dbb-839e-26a16ae439c9" />